File: problems/linked_lists/insertNode.js
const insertNode = (head, value, index) => {
  // todo
  if(index === 0) {
    let newNode = new Node(value)
      newNode.next = head  
      return newNode
  }
  if(index === 1) {
    const next = head.next 
    let newNode = new Node(value)
    head.next = newNode
    newNode.next = next 
  }
  insertNode(head.next, value, index - 1 )
  return head 

};

File: python_Structy/insert_node.py
class Node:
  def __init__(self, val):
    self.val = val
    self.next = None

# dummy -> a -> b -> c -> d


def insert_node(head, value, index):
  dummyHead = Node(None)
  dummyHead.next = head 
  prev = None 
  curr = dummyHead
  # next = dummyHead.next  

  while index >= 0:
    prev = curr 
    curr = curr.next 
    # next = curr.next 
    index-=1

  insert = Node(value)
  prev.next = insert
  insert.next = curr

  return dummyHead.next
